interfaces/portal/src/app/pages/projects-overview/components/create-project-dialog/create-project-dialog.component.ts

```diff
@@ -1,89 +1,229 @@
+  readonly formBudget = viewChild<ProjectFormBudgetComponent>('formBudget');
+
+  // 0 = dialog closed
+  // 1 = step 1: name
+  // 2 = step 2: information
+  // 3 = step 3: budget
+  readonly currentStep = signal<0 | 1 | 2 | 3>(0);
+
+  readonly formGroup = computed(() => {
+    const nameGroup = this.formName()?.formGroup;
+    const informationGroup = this.formInformation()?.formGroup;
+    const budgetGroup = this.formBudget()?.formGroup;
+
+    if (!nameGroup || !informationGroup || !budgetGroup) {
+      return undefined;
+    }
+
+    return new FormGroup({
+      nameGroup,
+      informationGroup,
+      budgetGroup,
+    });
   });
 
-  formFieldErrors = generateFieldErrors(this.formGroup);
+  readonly proceedLabel = computed(() =>
+    this.currentStep() !== 3
+      ? $localize`Continue`
+      : $localize`:@@create-project:Create project`,
+  );
 
   createProjectMutation = injectMutation(() => ({
-    mutationFn: ({
-      token,
-      assetId,
-    }: ReturnType<CreateProjectFormGroup['getRawValue']>) =>
-      this.projectApiService.createProjectFromKobo({ token, assetId }),
-    onSuccess: (project) => {
-      if (!project?.id) {
-        throw new Error($localize`No Project-ID returned.`);
-      }
+    mutationFn: async ({
+      nameGroup: { name, description },
+      informationGroup: {
+        startDate,
+        endDate,
+        enableScope,
+        location,
+        targetNrRegistrations,
+        validation,
+      },
+      budgetGroup: {
+        budget,
+        currency,
+        distributionDuration,
+        fixedTransferValue,
+      },
+    }: ReturnType<
+      FormGroup<{
+        nameGroup: ProjectNameFormGroup;
+        informationGroup: ProjectInformationFormGroup;
+        budgetGroup: ProjectBudgetFormGroup;
+      }>['getRawValue']
+    >) =>
+      this.projectApiService.createProject({
+        titlePortal: {
+          [LanguageEnum.en]: name,
+        },
+        description: {
+          [LanguageEnum.en]: description,
+        },
+        budget,
+        currency,
+        distributionDuration,
+        fixedTransferValue,
+        startDate: startDate ? startDate.toISOString() : undefined,
+        endDate: endDate ? endDate.toISOString() : undefined,
+        enableScope,
+        location,
+        targetNrRegistrations,
+        validation,
+      }),
+    onSuccess: async (result) => {
+      await this.projectApiService.invalidateCache();
+      await this.authService.refreshUser();
+
+      await this.router.navigate([
+        '/',
+        AppRoutes.project,
+        result?.id,
+        AppRoutes.projectSettings,
+      ]);
+
       this.toastService.showToast({
         detail: $localize`Project successfully created.`,
       });
-      return this.router.navigate([AppRoutes.project, project.id]);
+    },
+    onError: (error) => {
+      this.toastService.showToast({
+        severity: 'error',
+        detail: error.message,
+      });
     },
   }));
 
+  goBack() {
```

Comment:
If we're gonna use multistep forms a lot more I think it's good to make a better datastructure for this with some accompanying functions so we don't have to repeat this. This is the 2nd instance of it I believe.

Reply:
I reminded the dev-team in chat.